### DISCUSSION TITLE
best way to hide menu, eg , Run, Terminal etc

### DISCUSSION THREAD

**Discussion by icoco** [eclipse-theia/theia#13761] [Q&A] [answered]
after research, seem not official solution,  is it possible set the menu via json configuration file ? 

another question is, how to intercept or listener the menu /action event ? eg, while file opened or closed event

**Comment by msujew** [comment]
Hey @icoco,

> after research, seem not official solution, is it possible set the menu via json configuration file ?

There are "official" solution to this: if you only want to hide the menu entries, you can just call `MenuModelRegistry#unregisterMenu` with the appropriate menu path. If you want to completely disable those features, I would recommend you [our documentation on contribution filters](https://theia-ide.org/docs/contribution_filter/).

> another question is, how to intercept or listener the menu /action event ? eg, while file opened or closed event

For newly opened/closed editor widgets, you can hook into the [`EditorManager#onCreated`](https://github.com/eclipse-theia/theia/blob/a6fb8829705796ee43cf05aa220a61137e01c369/packages/editor/src/browser/editor-manager.ts#L69) event or similar events surrounding the addition/removal of widgets. Note that Theia doesn't really care about the underlying file source of a widget.

As for everything else: Read the code, the framework is pretty vast and offers basically limitless options to hook into events or modify behavior.

**Comment by icoco** [comment, marked as answer]
thanks for reply...

It works : 
```
        menus.unregisterMenuAction(CommonMenus.HELP[CommonMenus.HELP.length - 1],  MAIN_MENU_BAR)
     
        menus.unregisterMenuAction(CommonMenus.FILE_OPEN[CommonMenus.FILE_OPEN.length - 1])
        menus.unregisterMenuAction("workspace:saveAs")
        menus.unregisterMenuAction("workspace:addFolder")
        menus.unregisterMenuAction("workspace:close") 
```
